libnavigation-base/src/main/java/com/mapbox/navigation/base/route/model/RouteLegsNavigation.kt

```diff
@@ -1,5 +1,8 @@
 package com.mapbox.navigation.base.route.model
 
-data class RouteLegsNavigation(
-    val legs: List<*>
+data class RouteLegsNavigation constructor(
```

Comment:
I just followed some existing `data classes` naming convention. I don't like adding `Navigation` suffix, but since some classes were already defined this way I chose to keep the parity in naming.

Reply:
I agree, the `Navigation` suffix seems excessive and possibly we can drop it before releasing.

This comment was only about the plural, this class holds only one leg's data rather than a list of legs like it did before, so I would suggest renaming to `RouteLegNavigation` from `RouteLegsNavigation`.

Reply:
Actually, why are we removing the wrapper / encapsulation? 🤔 Normally, it's a good practice to wrap collections into their own objects. It allows you to give better naming to the objects with semantics to your domain and therefore more context.